### DISCUSSION TITLE
直链下载地址是否可配置

### DISCUSSION THREAD

**Discussion by leveljava** [AlistGo/alist#7121] [General]
alist通过nginx部署
目前通过ip访问 , 下载地址为   ip/alist/xxx.xxx
通过域名访问      下载地址为    域名/alist/xxx.xxx

alist和nginx不在同一台机器上 , 目前想实现 , 通过域名访问alist
获取的下载直链还可以是   ip/alist/xxx.xxx

**Comment by Simonzhang8** [comment]
三种方式可以试一下（没试过不确定是否可行）：
1、config.json中site_url填ip地址（你如果有二级目录也需要相应扩充）
2、nginx反代配置中proxy_set_header Host $http_host;的$http_host修改为ip地址
3、存储配置里下载代理地址填ip

**Comment by leveljava** [comment]
1,3都试过了 , 不行
就类似于 https://alist.pages.dev/中 获取的下载直链是
https://al.nn.ci/d/alist.png?sign=xX-A70KA2cNxomfwvFsJORrQJDK7J8xiuqC_1CwJu6Y=:0

**Reply by anwen-anyi** [reply]
> 1,3都试过了 , 不行 就类似于 [https://alist.pages.dev/中](https://alist.pages.dev/%E4%B8%AD) 获取的下载直链是 https://al.nn.ci/d/alist.png?sign=xX-A70KA2cNxomfwvFsJORrQJDK7J8xiuqC_1CwJu6Y=:0

配置文件的 `site_url` 配置一下呢

**Reply by Simonzhang8** [reply]
> > 1,3都试过了 , 不行 就类似于 [https://alist.pages.dev/中](https://alist.pages.dev/%E4%B8%AD) 获取的下载直链是 https://al.nn.ci/d/alist.png?sign=xX-A70KA2cNxomfwvFsJORrQJDK7J8xiuqC_1CwJu6Y=:0
> 
> 配置文件的 `site_url` 配置一下呢

那不就是1吗😂

**Reply by Simonzhang8** [reply]
> 1,3都试过了 , 不行 就类似于 [https://alist.pages.dev/中](https://alist.pages.dev/%E4%B8%AD) 获取的下载直链是 https://al.nn.ci/d/alist.png?sign=xX-A70KA2cNxomfwvFsJORrQJDK7J8xiuqC_1CwJu6Y=:0

3试了下确实不行，是我对代理地址的逻辑理解错了
1不可以吗？正常1应该是没问题的呀😂 你那边修改config.json以后下载地址是什么样的啊？

**Reply by anwen-anyi** [reply]
> > > 1,3都试过了 , 不行 就类似于 [https://alist.pages.dev/中](https://alist.pages.dev/%E4%B8%AD) 获取的下载直链是 https://al.nn.ci/d/alist.png?sign=xX-A70KA2cNxomfwvFsJORrQJDK7J8xiuqC_1CwJu6Y=:0
> > 
> > 
> > 配置文件的 `site_url` 配置一下呢
> 
> 那不就是1吗😂

忘记了前文，回复了个寂寞 哈哈 好尴尬...

**Reply by leveljava** [reply]
修改了都无变化

**Comment by cai12121** [comment]
你好, 我也碰到这个问题, 有解决方案吗? 
域名访问时的下载地址: https://xxxx.xxxx.cn/d/%E7%99%BE%E5%BA%A6%E7%BD%9..... 
本地IP访问时的下载地址: http://192.168.xx.xx:5244/d/%E7%99%BE%E5%BA%A6%E7%BD%9....
试过1也不行, 下载地址还是https://xxxx.xxxx.cn/d/%E7%99%BE%E

**Reply by cai12121** [reply]
"site_url": "http://192.168.xx.xx:5244",
